### ISSUE TITLE
Multiple exceptions in drakparse.py. (drak-postprocess)

### ISSUE THREAD

**Issue by vornick** [CERT-Polska/drakvuf-sandbox#524]
Hi, I noticed a few exceptions in drakparse.py.
1) Key missing in procmon dict
```
ERROR:root:BUG: Failed to parse log entry.
{"Plugin":"procmon","TimeStamp":"1619087668.782024","PID":4,"PPID":0,"RunningProcess":"System"}
Traceback (most recent call last):
  File "D:\Python_code\drakparse.py", line 178, in parse_logs
    converted = str(plugin_obj(line_obj))
  File "D:\Python_code\drakparse.py", line 74, in __init__
    if obj["Method"] == "NtCreateUserProcess":
KeyError: 'Method'
```
I solved this to next code(add key.get check):
```
class Procmon(Base):
    def __init__(self, obj: Dict):

        if obj.get("Method") != None:
            if obj["Method"] == "NtCreateUserProcess":
                super().__init__(obj, "Process Create", obj["ImagePathName"], detail=f'PID: {obj["NewPid"]}, Command line: {obj["CommandLine"]}')
```
2) Timestamp error format string:
```
ERROR:root:BUG: Failed to parse log entry.
{"Plugin":"syscall","TimeStamp":"1619013174.936639","PID":636,"PPID":496,"TID":2844,"UserName":"SessionID","UserId":0,"ProcessName":"\\Device\\HarddiskVolume2\\Windows\\System32\\svchost.exe","Method":"NtCreateThreadEx","EventUID":"0x2a0d1b","Module":"nt","vCPU":1,"CR3":"0x8b7d7000","Syscall":165,"NArgs":11,"ThreadHandle":"0xfffff880047437e0","DesiredAccess":"0x1fffff","ObjectAttributes":"0xfffff880047437e8","ProcessHandle":"0xffffffff800002b8","StartRoutine":"0x7754fbc0","Argument":"0x22dfd0","CreateFlags":"0x1","ZeroBits":"0x0","StackSize":"0x8000","MaximumStackSize":"0x80000","AttributeList":"0xfffff88004743830"}
Traceback (most recent call last):
  File "D:\Python_code\drakparse.py", line 177, in parse_logs
    converted = str(plugin_obj(line_obj))
  File "D:\Python_code\drakparse.py", line 110, in __init__
    super().__init__(obj, "Thread Create", "")
  File "D:\Python_code\drakparse.py", line 13, in __init__
    self.timestamp = datetime.utcfromtimestamp(float(obj["TimeStamp"])).strftime("%-I:%-M:%-S.%f %p")
ValueError: Invalid format string
```
In my case, it is fixed like this:
before:
self.timestamp = datetime.utcfromtimestamp(float(obj["TimeStamp"])).strftime(**"%-I:%-M:%-S.%f %p"**)
after:
self.timestamp = datetime.utcfromtimestamp(float(obj["TimeStamp"])).strftime("**%I:%M:%S.%f %p"**)

3)  String does not check for incorrect character before json.loads(line) work:
Its similar error for all lines with suspicious character:
```
Traceback (most recent call last):
  File "D:\Python_code\drakparse.py", line 164, in parse_logs
    line_obj = json.loads(line)
  File "C:\Python38\lib\json\__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "C:\Python38\lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Python38\lib\json\decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Invalid control character at: line 1 column 332 (char 331)
ERROR:root:BUG: Unparseable log entry!
```
lines:
```
{"Plugin":"syscall","TimeStamp":"1619087671.776687","PID":1476,"PPID":1224,"TID":1896,"UserName":"SessionID","UserId":1,"ProcessName":"\\Device\\HarddiskVolume2\\Windows\\explorer.exe","Method":"NtQueryValueKey","EventUID":"0x1190a","Module":"nt","vCPU":1,"CR3":"0x6ca1f000","Syscall":20,"NArgs":6,"KeyHandle":"0x60c","ValueName":"ma=OI5!!!!4!!!!ZXXFxFrgYnathntrSvyrf<","KeyValueInformationClass":"0x2","KeyValueInformation":"0x4bbcbb0","Length":"0x90","ResultLength":"0x4bbcb64"}
{"Plugin":"syscall","TimeStamp":"1619087728.838131","PID":1060,"PPID":496,"TID":1360,"UserName":"SessionID","UserId":0,"ProcessName":"\\Device\\HarddiskVolume2\\Windows\\System32\\SearchIndexer.exe","Method":"NtQueryInformationFile","EventUID":"0x1292c1","Module":"nt","vCPU":1,"CR3":"0x875b9000","Syscall":14,"NArgs":5,"FileHandle":"б’¦","IoStatusBlock":"0x438e000","FileInformation":"0x438e090","Length":"0x104","FileInformationClass":"0x9"}
{"Plugin":"syscall","TimeStamp":"1619087728.837952","PID":1060,"PPID":496,"TID":1360,"UserName":"SessionID","UserId":0,"ProcessName":"\\Device\\HarddiskVolume2\\Windows\\System32\\SearchIndexer.exe","Method":"NtCreateFile","EventUID":"0x1292bd","Module":"nt","vCPU":1,"CR3":"0x875b9000","Syscall":82,"NArgs":11,"FileHandle":"0x438e028","DesiredAccess":"0x100080","ObjectAttributes":"\\б’¦","IoStatusBlock":"0x438dfd0","AllocationSize":"0x0","FileAttributes":"0x0","ShareAccess":"0x7","CreateDisposition":"0x1","CreateOptions":"0x2020","EaBuffer":"0x0","EaLength":"0x0"}
```
As you can see incorrect characters here:
"ValueName":"ma=OI5!!!!4!!!!ZXXFxFrgYnathntrSvyrf<",
"FileHandle":"б’¦",
"ObjectAttributes":"\\б’¦",
I have no idea how to check invalid characters here:
```
for line in lines:
    try:
        line_obj = json.loads(line)
    except Exception:
        logging.exception(f"BUG: Unparseable log entry!\n{line}")
        continue
```

Drakmon.log is here https://disk.yandex.ru/d/IR8nU9UVfYn-hg
I hope you can provide me some tips on how to correctly fix these errors. Especially number 3.

**Comment by BonusPlay** [comment]
This will hopefully be fixed in 0.16 release, as https://github.com/tklengyel/drakvuf/commit/7d9456adcb791ca9892822dcb051a69e7ee9334f was merged in drakvuf.

**Comment by chivay** [comment]
Shoud be fixed by bumping DRAKVUF.
